### DISCUSSION TITLE
device tracker incorrectly transitioning car from not_home to home during home assistant restart

### DISCUSSION THREAD

**Discussion by dblaber** [teslamate-org/teslamate#3065] [Q&A]
Hi, I have this problem with home assistant restarts (due to upgrade, power outage etc) that my garage door pops open. This is because device tracker is defaulting to not home after a restart, and transitioning to home, of which my automation rule (based on example) is triggered.  Has anyone figured a way to avoid this behavior? 

I think its because MQTT topics are empty initially after a restart, hence reporting its not home, and after they get populating from teslamate, it transitions back.  I was thinking of setting default gps coorindates for mqtt sensor topic  (https://www.home-assistant.io/integrations/sensor.mqtt/) , however doesn't feel right doing that. I looked at history of Geofence sensor as well, and that seems to not go into 'not home' when device is restarting. Anyone use geofence sensor for garage automation rules?

**Comment by vincep5** [comment]
The device_tracker and the geofence mqtt sensor both can go "unavailable" during a reboot and thus making an automation think that the vehicle has arrived home.  What I tried doing was making sure that a few things were true in the Automation's conditions:
Garage door is closed
mqtt topic of **state** is "driving"

Maybe check that both the device_tracker and geofence are both home?

I've disabled my automation, I'm worried that it could open my garage door when I'm already home.  Or if I drive past my home and go onto another destination.

**Comment by gmichels** [comment]
I had the same issue, and what I've done to mitigate it is add a condition to the automation to ensure the car speed is greater than 0.
